### ISSUE TITLE
Pinned Dependencies doc questions

### ISSUE THREAD

**Issue by frankychung** [rescript-association/rescript-lang.org#397]
I followed along with the example on the [Pinned Dependencies page](https://rescript-lang.org/docs/manual/latest/build-pinned-dependencies) and had two questions.

When trying what is written in the last paragraphs:

> Now, whenever we are running `rescript build` within our `app` package, the compiler would always rebuild any changes within its pinned dependencies as well.

After a `rescript clean`, `rescript`/`rescript build` worked fine. But if I made further changes in any of the pinned dependencies, running the normal build in the `app` package would not reflect those changes. In other words, with the normal `build`, only changes directly in the `app` workspace were compiled. However, if I ran `rescript build -with-deps` in the toplevel `app` workspace, changes in all pinned dependencies compiled fine. Is this the expected workflow and if so, should we add this point to these docs? After figuring out needing `-with-deps`, using my own watcher at the toplevel package to run this as needed is a great workflow for my project.

Second, in the [codebase example snippet with Yarn Workspaces](https://rescript-lang.org/docs/manual/latest/build-pinned-dependencies#yarn-workspaces), I had to make a `package.json` in workspace as well before running `yarn install`, otherwise rescript would give me an error that the dependency is not installed. Would it be worth adding this to the snippet to make it easier to follow along with the example?

**Comment by ryyppy** [comment]
Good input. I guess `--with-deps` is necessary here...
@bobzhang what do you think? 

> Second, in the codebase example snippet with Yarn Workspaces, I had to make a package.json in workspace as well before running yarn install, otherwise rescript would give me an error that the dependency is not installed. Would it be worth adding this to the snippet to make it easier to follow along with the example?

You mean each subpackage should list its own package.json? Sounds like an adaption we want

**Comment by Minnozz** [comment]
See [here](https://github.com/Minnozz/rescript-reproduce-issue/tree/pinned-dependencies-issue) for an example Yarn workspaces repository I created for https://github.com/rescript-lang/rescript-compiler/issues/5197 (which is somewhat related).

**Comment by frankychung** [comment]
> You mean each subpackage should list its own package.json?

Yeah, it's a detail about workspaces and not rescript, but still a gotcha when following the example

**Comment by fhammerschmidt** [comment]
Pinned deps are obsolete in v12
